apps/server/src/shared/domain/rules/submission.rule.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable } from '@nestjs/common';
+import { Submission, User } from '../entity';
+import { IPermissionContext, PermissionTypes } from '../interface/permission';
+import { Actions } from './actions.enum';
+import { BasePermission } from './base-permission';
+import { CourseGroupRule } from './course-group.rule';
+import { TaskRule } from './task.rule';
+
+@Injectable()
+export class SubmissionRule extends BasePermission<Submission> {
```

Comment:
Wo stellen wir denn sicher, dass eine Lehrer nicht die Abgabe verändert und ein Schüler nicht die Bewertung? Das wird erst mal weiterhin über den Controller in Feathers geregelt?

Reply:
Wird aktuell durch den Legacy Controller geregelt. Durch die Entscheidung das Grading in der Submission zu belassen, wird es auch schwer dies über die Submission Rule abzubilden. Durch eine Aufteilung in Submission und Grading könnte es eine eigene Grading Rule geben, die genau das regelt. Die Alternative wäre dann dieses Permission Handling im UC abzubilden. Aber so wie ich es im Refinement verstanden habe ist es einfach tbd? Und betrifft diesen PR nicht oder?

Reply:
Das sind auch meiner Meinung genau die Schwachstellen dieser Umsetzung: Einerseits, dass Submission und Grading nicht getrennt behandelt werden können, und andererseits, dass die string-based Permissions vom UC gehandlet werden. Aber wir haben uns jetzt darauf geeinigt, das erstmal so umzusetzen.